cmd/cloudshell_open/deploy.go

```diff
@@ -82,6 +82,20 @@ func optionsToResourceRequirements(options options) *runapi.ResourceRequirements
 	return &runapi.ResourceRequirements{Limits: limits}
 }
 
+func optionsToContainerSpec(options options) *runapi.ContainerPort {
+	var containerPortName = "http1"
+	if options.Http2 != nil && *options.Http2 {
+		containerPortName = "h2c"
+	}
+
+	var containerPort = 8080
```

Comment:
I'll check with the team if this is allowed:

```
ports:
- name: h2c
```

which doesn't specify PORT and lets system assign 8080.

in Kubernetes this is not allowed, but on Cloud Run it might be.

otherwise we'd be hardcoding 8080, which actually isn't part of our runtime/container contract. so if Cloud Run changes 8080 default to something else tomorrow we'll be diverging from that.

Reply:
From what I could tell looking at Cloud Run YAML is that `8080` was the default value for all services. But yeah, I'm not sure how that translates to using `PORT`.